pkg/scheduler/testing/utils.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+// WantStatusError is used to compare status with errors in tests.
+type WantStatusError struct {
+	Err error
+}
+
+func (e WantStatusError) Error() string {
+	return e.Err.Error()
+}
+
+func (e WantStatusError) Is(target error) bool {
```

Comment:
This is ultimately comparing strings, which is going around the problem.

We don't want to compare strings. You can either compare the pointer to the same error or you can type the actual error.

Reply:
> This is ultimately comparing strings, which is going around the problem.
> 
> We don't want to compare strings. You can either compare the pointer to the same error or you can type the actual error.

How about this ?

```
func (e WantStatusError) Is(target error) bool {
	return errors.Is(e.Err, target) || e.Error() == target.Error()
}
```

We will import errors.

Reply:
Having a `WantStatusError` is not appropriate. The error that should be typed is the one generated by the production code, not the test code. Another alternative is to have an exported error from the production code. Then we can just compare the pointer to the error.

Reply:
> Having a `WantStatusError` is not appropriate. The error that should be typed is the one generated by the production code, not the test code. Another alternative is to have an exported error from the production code. Then we can just compare the pointer to the error.

So we still consider comparing pointers instead of implementing `Is()`.

Reply:
`cmp.Diff` will compare pointers or use `Is`

If you implement `Is`, it should be in the production code, not the test code.